### DISCUSSION TITLE
1.1.12

### DISCUSSION THREAD

**Discussion by koriym** [koriym/Koriym.PhpSkeleton#74] [Announcements]
* Better PHP8 support

<hr /><em>This discussion was created from the release <a href='https://github.com/koriym/Koriym.PhpSkeleton/releases/tag/1.1.12'>1.1.12</a>.</em>

**Comment by kenjis** [comment]
The config for path coverage is gone in 1.1.12.
This commit https://github.com/koriym/Koriym.PhpSkeleton/pull/69/commits/0a3c4f9cc4d1b87fdd56074fb2aa92f77ef7c25e is gone.

```diff
$ git diff 1.1.11...1.1.12 phpunit.xml.dist
diff --git a/phpunit.xml.dist b/phpunit.xml.dist
index 96582b0..f92548e 100644
--- a/phpunit.xml.dist
+++ b/phpunit.xml.dist
@@ -1,20 +1,20 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-  xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd"
-  bootstrap="vendor/autoload.php">
-  <coverage cacheDirectory=".phpunit.cache/code-coverage"
-    processUncoveredFiles="true"
-    pathCoverage="true">
-    <include>
-      <directory suffix=".php">src</directory>
-    </include>
-  </coverage>
-  <testsuites>
-    <testsuite name="all">
-      <directory>tests</directory>
-    </testsuite>
-  </testsuites>
-  <php>
-    <ini name="error_reporting" value="-1"/>
-  </php>
+<phpunit
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:noNamespaceSchemaLocation="./vendor/phpunit/phpunit/phpunit.xsd"
+    bootstrap="vendor/autoload.php"
+>
+    <testsuites>
+        <testsuite name="__Vendor__.__Package__ test suite">
+            <directory>tests</directory>
+        </testsuite>
+    </testsuites>
+    <coverage processUncoveredFiles="true">
+        <include>
+            <directory suffix=".php">src</directory>
+        </include>
+    </coverage>
+    <php>
+        <ini name="error_reporting" value="-1"/>
+    </php>
 </phpunit>
```

**Reply by kenjis** [reply]
@koriym It seems the change commit is https://github.com/koriym/Koriym.PhpSkeleton/pull/72/commits/953cf640f1e9a209bacf3088da363e3de3d8af23 , but there is no `pathCoverage` in the diff.
It is strange.

**Reply by kenjis** [reply]
This merge commit overrided my changes?
<img width="956" alt="Screenshot 2021-07-15 11 30 57" src="https://user-images.githubusercontent.com/87955/125718830-44706c59-8603-4eca-ad52-4383d1481bf3.png">

**Comment by koriym** [comment]
Yes, I think I've done something wrong.
However, I don't think `pathCoverage="true"` is necessary in the default options.

Only xdebug is supported, and I think it's better to keep it simple by default.

**Reply by kenjis** [reply]
> Only xdebug is supported

It's very bad news!

I think line coverage is not enough. But It's hard to make it the default config.

**Reply by koriym** [reply]
phpdbg doesn't work even for line coverage with pathCoverage=1

```
% phpdbg -qrr ./vendor/bin/phpunit --coverage-htm coverage
PHPUnit 9.5.6 by Sebastian Bergmann and contributors.

Warning:       No code coverage driver with path coverage support available

............................                                      28 / 28 (100%)

Time: 00:00.059, Memory: 18.00 MB
```

**Comment by koriym** [comment]
If the package supports only Xdebug coverage (not this package), it's fairly easy to set up.
Shouldn't be a problem.
